agent/credentials/manager.go

```diff
@@ -68,14 +68,10 @@ type IAMRoleCredentials struct {
 type TaskIAMRoleCredentials struct {
 	ARN                string
 	IAMRoleCredentials IAMRoleCredentials
-	lock               sync.RWMutex
```

Comment:
note: This lock is likely redundant.  There's a taskCredentialsLock which is managing the writes and reads and the usage for this lock is only on reads but is covered by the other outer lock.  Approving